DataMigration/src/main/java/org/sagebionetworks/dian/datamigration/tools/schedulev2/ScheduleV2Migration.java

```diff
@@ -60,58 +63,65 @@ public static void runV2Migration() throws IOException {
 
             // Once we are ready to deploy this for all studies, use a curated list of all Study IDs
             for (String arcId : arcIdList) {
-                StudyParticipant p = BridgeJavaSdkUtil.getParticipantByExternalId(arcId);
-
-                if (p.getStudyIds() == null || p.getStudyIds().isEmpty()) {
-                    System.out.println(arcId + " has withdrawn");
-                    continue; // user has withdrawn, no need to migrate
-                }
-
-                String uId = p.getId();
-                String sId = p.getStudyIds().get(0);
-
-                SageV2Availability availability = createV2Availability(
-                        uId, getAvailabilityJsonFromBridge(uId));
-                if (availability == null) {
-                    System.out.println(arcId + " has no availability");
-                    continue;
-                }
-
-                SageV1Schedule v1Schedule = createV1Schedule(uId, getScheduleJsonFromBridge(uId));
-                if (v1Schedule == null) {
-                    System.out.println(arcId + " has no schedule");
-                    continue;
-                }
-
-                System.out.println("Performing V2 migration on " + arcId);
-
-                // Check for a user that has already migrated to V2 and signed into the app
-                SageUserClientData clientData = SageUserClientData.Companion.fromStudyParticipant(gson, p);
-                boolean hasMigrated = SageUserClientData.Companion.hasMigrated(clientData);
-
-                if (!hasMigrated) {
-                    System.out.println("Creating schedule for " + arcId);
-                    createV2Schedule(v1Schedule, uId, sId);
```

Comment:
I wrapped each participant's migration in a try/catch where errors are collected and then printed at the end of the program execution.  This allows for the migration to fully operate even if there is a single user error.